internal/commands/app/create_inputs_test.go

```diff
@@ -875,6 +875,366 @@ func TestAppCreateInputsResolveCluster(t *testing.T) {
+			newAppInputs:                   newAppInputs{Name: "test-app"},
+			ServerlessInstances:            []string{"test-serverless-instance"},
+			ServerlessInstanceServiceNames: []string{"mongodb-atlas"},
+		}
+
+		ds, _, err := inputs.resolveServerlessInstances(ui, ac, "123")
+		assert.Nil(t, err)
+
+		assert.Equal(t, []dataSourceCluster{
+			{
+				Name: "mongodb-atlas",
+				Type: realm.ServiceTypeCluster,
+				Config: configCluster{
+					ClusterName:         "test-serverless-instance",
+					ReadPreference:      "primary",
+					WireProtocolEnabled: false,
+				},
+				Version: 1,
+			},
+		}, ds)
+		assert.Equal(t, "123", expectedGroupID)
+	})
+
+	t.Run("should return data source config of multiple provided serverless instances", func(t *testing.T) {
+		_, ui := mock.NewUI()
+		var expectedGroupID string
+		ac := mock.AtlasClient{}
+		ac.ServerlessInstancesFn = func(groupID string) ([]atlas.ServerlessInstance, error) {
+			expectedGroupID = groupID
+			return []atlas.ServerlessInstance{
+				{Name: "test-serverless-instance-1"},
+				{Name: "test-serverless-instance-2"},
+			}, nil
+		}
+
+		inputs := createInputs{
+			newAppInputs:                   newAppInputs{Name: "test-app"},
+			ServerlessInstances:            []string{"test-serverless-instance-1", "test-serverless-instance-2"},
+			ServerlessInstanceServiceNames: []string{"mongodb-atlas", "another-data-source"},
+		}
+
+		ds, _, err := inputs.resolveServerlessInstances(ui, ac, "123")
+		assert.Nil(t, err)
+
+		assert.Equal(t, []dataSourceCluster{
+			{
+				Name: "mongodb-atlas",
+				Type: realm.ServiceTypeCluster,
+				Config: configCluster{
+					ClusterName:         "test-serverless-instance-1",
+					ReadPreference:      "primary",
+					WireProtocolEnabled: false,
+				},
+				Version: 1,
+			},
+			{
+				Name: "another-data-source",
+				Type: realm.ServiceTypeCluster,
+				Config: configCluster{
+					ClusterName:         "test-serverless-instance-2",
+					ReadPreference:      "primary",
+					WireProtocolEnabled: false,
+				},
+				Version: 1,
+			},
+		}, ds)
+		assert.Equal(t, "123", expectedGroupID)
+	})
+
+	t.Run("should warn user if serverless instances are not found and auto confirm is set", func(t *testing.T) {
+		profile, teardown := mock.NewProfileFromTmpDir(t, "app_create_test")
+		defer teardown()
+
+		console, ui, err := mock.NewConsoleWithOptions(mock.UIOptions{AutoConfirm: true})
+		assert.Nil(t, err)
+		defer console.Close()
+
+		testApp := realm.App{
+			ID:          primitive.NewObjectID().Hex(),
+			GroupID:     primitive.NewObjectID().Hex(),
+			ClientAppID: "test-app-abcde",
+			Name:        "test-app",
+		}
+
+		rc := mock.RealmClient{}
+		rc.CreateAppFn = func(groupID, name string, meta realm.AppMeta) (realm.App, error) {
+			return testApp, nil
+		}
+		rc.ImportFn = func(groupID, appID string, appData interface{}) error {
+			return nil
+		}
+		rc.AllTemplatesFn = func() ([]realm.Template, error) {
+			return []realm.Template{}, nil
+		}
```

Comment:
These are actually needed because of [this call](https://github.com/10gen/realm-cli/blob/b499a5e798e59675648216f6778f290e2c97b189/internal/commands/app/create_inputs_test.go#L502), which calls the app create command handler.

`AllTemplatesFn` isn't actually needed though because we're not inputting a template id, so I'll remove that